TELCODOCS-1476 hub-side templating (#68139)
openshift/openshift-docs: modules/ztp-deploying-siteconfig-hub-side-template.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-sites.adoc

:_mod-docs-content-type: PROCEDURE
[id="ztp-deploying-siteconfig-hub-side-template_{context}"]

> = Deploying a SiteConfig with hub-side templating

Reviewer: This could be interpreted that the SiteConfig is being templated, which is not. Maybe a rephrase would help here, something like "Deploying SiteConfig to support hub templated PolicyGenTemplates" or something along these lines.

Author: good point. I'd propose: 
"Deploying a managed cluster with `SiteConfig` and `PolicyGenTemplates` using hub-side templates"
WDYT?

Reviewer: "Deploying a managed cluster with SiteConfig where PolicyGenTemplates is using hub-side templates"?

> The following is an example `SiteConfig` custom resource (CR) that uses the `spec.clusters.siteConfigMap.data` field to specify per-site data to be used in hub side templating.

Reviewer: ```suggestion
The following is a `SiteConfig` custom resource (CR) example that includes the new optional `spec.clusters.siteConfigMap`  where the `data` field contains per-site data to be used in hub side templating.
```

Author: There can be an issue with referring to a feature as "new" or "optional" in product documentation because, as time passes, the feature ceases to be new but the documentation won't be updated to reflect that. Release Notes are the best place to point out the newness of a feature.
I've rephrased though, in order to draw attention to the `spec.clusters.siteConfigMap` section and mention the `data` field separately

Reviewer: Ok, but I think we should still say that it's an optional section, it's not mandatory. :thinking:

.Example {sno} `SiteConfig` CR with hub-side templating
[source,yaml]
----
include::snippets/ztp_example-siteconfig-hub-side-templating.yaml[]
----

> [NOTE]

Reviewer: Maybe the section below can be a bit restructured to something like:

>Maybe all the section above can be a bit restructured to something like:
Besides the `data` field, the `siteConfigMap` section also includes the namespace where the `ConfigMap` is to be created and its name. 

For a successful integration with the PolicyGenTemplate, the `siteConfigMap.namespace` you specify must be the same as the namespace used by the `PolicyGenTemplate` resources.

If not specified, `siteConfigMap.namespace` will default to <ztp-site>, and `siteConfigMap.namespace` to ztp-site-<cluster_name>.

Author: Maybe we could have another look at this together?

====

> The generated `ConfigMap` CRs follow the naming convention `ztp-site-<cluster_name>-configMap` and are created under the same namespace as the policy resource.

Reviewer: Maybe just leave
> The generated `ConfigMap` CRs follow the naming convention `ztp-site-<cluster_name>`.
since you're explaining the namespace aspect below?

Author: ok. I removed reference to the namespace and left it at `ztp-site-<cluster_name>-configMap`.

Reviewer: It should be `ztp-site-<cluster-name>` though.

> The `siteConfigMap` section includes the namespace where the `ConfigMap` is to be created.

Reviewer: ```suggestion
The `siteConfigMap` section includes the namespace where the `ConfigMap` is to be created and its name. 
```

Author: sorry @irinamihai I'm not clear on what the "its" refers to here.
Is "its name" the name of the namespace or the name of the `ConfigMap`?
The "name of the ConfigMap" doesn't seem right but the "name of the namespace" sounds kinda off too.

Author: name refers to ConfigMap - add this to the example yaml 
Something like
SiteConfigMap
name: site-example-sno

> The `siteConfigMap.namespace` you specify must be the same as the namespace used by the `PolicyGenTemplate` resources.

Reviewer: ```suggestion
For a successful integration with the PolicyGenTemplate, the `siteConfigMap.namespace` you specify must be the same as the namespace used by the `PolicyGenTemplate` resources.
```
